core/auth/credentials.go

```diff
@@ -51,47 +53,56 @@ func NewCredentials(username, password string, db Storage) *Credentials {
 	}
 }
 
-// Validate username and password against stored Credentials
+// Validate username and password against stored Credentials.
 func (credentials *Credentials) Validate() (err error) {
-	var storedHash string
-	storedHash, err = credentials.loadOrInitialize()
-	if err != nil {
-		return errors.Wrap(err, "could not load credentials")
+	if credentials.username != config.FlagTequilapiUsername.Value {
+		return ErrBadCredentials
 	}
-	if credentials.username != username {
-		return errors.New("bad credentials")
+
+	storedHash, err := credentials.loadOrInitialize()
+	if err != nil {
+		return fmt.Errorf("could not load credentials: %w", err)
 	}
+
 	err = bcrypt.CompareHashAndPassword([]byte(storedHash), []byte(credentials.password))
 	if err != nil {
-		return errors.Wrap(err, "bad credentials")
+		return fmt.Errorf("bad credentials: %w", err)
 	}
+
 	return nil
 }
 
 func (credentials *Credentials) loadOrInitialize() (s string, err error) {
 	var storedHash string
-	err = credentials.db.GetValue(credentialsDBBucket, username, &storedHash)
-	if err == storage.ErrNotFound {
+
+	err = credentials.db.GetValue(credentialsDBBucket, config.FlagTequilapiUsername.Value, &storedHash)
+	if errors.Is(err, storage.ErrNotFound) {
```

Comment:
nitpick: While we're here we could invert this `if` moving out the code block outside. This would produce more readable code imo.